### DISCUSSION TITLE
Expose `useModalState` hook

### DISCUSSION THREAD

**Discussion by JackCA** [rainbow-me/rainbowkit#672] [Ideas]
it would be nice to export the [`useModalState` ](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/components/RainbowKitProvider/ModalContext.tsx#L127-L136)hook from the library's index -- I have the code change but wanted to open this before a PR

This would enable a developer to know whether or not the user abandoned a certain flow

Related to: https://github.com/rainbow-me/rainbowkit/discussions/624

**Comment by saihaj** [comment]
I mean there are hooks to open those modals which should solve any use case https://www.rainbowkit.com/docs/modal-hooks

**Reply by JackCA** [reply]
Those only open the modals, they don't tell you if they are currently open or not
